### ISSUE TITLE
[bug] boost 1.85.0 compilation errors 

### ISSUE THREAD

**Issue by victorsk2019** [dogecoin/dogecoin#3548]
Hi all,

I came across recently with various compilation issues with new boost 1.85.0 version.  This bug report is more of a "heads up" because it may be a while before this version will be used as a dependency for Dogecoin Core production, however, I think this is something we may have to deal with eventually.   I've upgraded Ubuntu's boost 1.74.0 to 1.85.0 for testing and because boost 1.85.0 is already available in Gentoo Linux.

`grep "define BOOST_VERSION " version.hpp 
#define BOOST_VERSION 108500`

Please see below errors (and possible fixes) when trying to compile Dogecoin Core 1.15.0 with boost 1.85.0 using this configure command:

`./configure --bindir=/home/m3t00/projects/fromgit/development/1.15.0/1.15.0-bin --disable-tests --disable-bench --with-gui=qt5 --disable-zmq --enable-c++14`

**Current behavior:**
Error 1:
```  CXX      libdogecoin_server_a-blockencodings.o
  CXX      libdogecoin_server_a-chain.o
In file included from chain.cpp:8:
validation.h:337:53: error: ‘std::list’ has not been declared
  337 |                         bool* pfMissingInputs, std::list<CTransactionRef>* plTxnReplaced = NULL,
      |                                                     ^~~~
validation.h:337:57: error: expected ‘,’ or ‘...’ before ‘<’ token
  337 |                         bool* pfMissingInputs, std::list<CTransactionRef>* plTxnReplaced = NULL,
      |                                                         ^
validation.h:342:74: error: ‘std::list’ has not been declared
  342 |                         bool* pfMissingInputs, int64_t nAcceptTime, std::list<CTransactionRef>* plTxnReplaced = NULL,
      |                                                                          ^~~~
validation.h:342:78: error: expected ‘,’ or ‘...’ before ‘<’ token
  342 |                         bool* pfMissingInputs, int64_t nAcceptTime, std::list<CTransactionRef>* plTxnReplaced = NULL,
      |                                                                              ^
make[2]: *** [Makefile:5971: libdogecoin_server_a-chain.o] Error 1
make[2]: Leaving directory '/home/m3t00/projects/fromgit/development/1.15.0/dogecoin/src'
make[1]: *** [Makefile:9848: all-recursive] Error 1
```
This error can be fixed by adding `#include <list>` to `validation.h`

Error 2:
```
 CXX      rpc/libdogecoin_util_a-protocol.o
rpc/protocol.cpp: In function ‘boost::filesystem::path GetAuthCookieFile()’:
rpc/protocol.cpp:76:15: error: ‘class boost::filesystem::path’ has no member named ‘is_complete’
   76 |     if (!path.is_complete()) path = GetDataDir() / path;
      |               ^~~~~~~~~~~
make[2]: *** [Makefile:6545: rpc/libdogecoin_util_a-protocol.o] Error 1
make[2]: Leaving directory '/home/m3t00/projects/fromgit/development/1.15.0/dogecoin/src'
make[1]: *** [Makefile:9848: all-recursive] Error 1
```
I fixed this error by replacing `is_complete()` function with `is_absolute()` because in older boost versions `is_complete()` returns `is_absolute()` implementation: ` bool is_complete() const { return is_absolute(); }` but in 1.85.0 they seem to just have `is_absolute()` instead.

Errors 3:
```
wallet/wallet.cpp: In static member function ‘static bool CWallet::Verify()’:
wallet/wallet.cpp:462:27: error: ‘basename’ is not a member of ‘fs’; did you mean ‘rename’?
  462 |     if (walletFile != fs::basename(walletFile) + fs::extension(walletFile))
      |                           ^~~~~~~~
      |                           rename
wallet/wallet.cpp:462:54: error: ‘extension’ is not a member of ‘fs’
  462 |     if (walletFile != fs::basename(walletFile) + fs::extension(walletFile))
      |                                                      ^~~~~~~~~
wallet/wallet.cpp: In member function ‘bool CWallet::BackupWallet(const string&)’:
wallet/wallet.cpp:4064:58: error: ‘fs::copy_option’ has not been declared
 4064 |                     fs::copy_file(pathSrc, pathDest, fs::copy_option::overwrite_if_exists);
      |                                                          ^~~~~~~~~~~
make[2]: *** [Makefile:6769: wallet/libdogecoin_wallet_a-wallet.o] Error 1
make[2]: Leaving directory '/home/m3t00/projects/fromgit/development/1.15.0/dogecoin/src'
make[1]: *** [Makefile:9848: all-recursive] Error 1
```
The error in line 4064, I think, is fixed by replacing `copy_option` with copy_option**s** because in `fs` namespace this enum is so declared.

Basically, many errors seem to get fixed by changing to different function or type variable names, depending on their corresponding implementation from previous boost versions.  

**Expected behavior**
Dogecoin Core should compile with `boost 1.85.0` without errors

**Steps to reproduce:**
Update to boost 1.85.0 and re-compile Dogecoin Core.

**Dogecoin Core version**
1.14.7 and 1.15.0

**Machine specs**
- OS: Ubuntu 22.04 (Jammy) (gcc 11) and Gentoo Linux (gcc 14)
- CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
- RAM: 16GB
- Disk size: 500GB
- Disk Type (HD/SDD): SSD

Hopefully, this report will give more information on what issues may be expected when installing Dogecoin Core with `boost 1.85.0`.  Thanks!
